modules/deploy/25-couchdb-deploy.erb

```diff
@@ -29,17 +31,19 @@ spec:
             path: /healthz
             port: 8080
           initialDelaySeconds: 5
-          timeoutSeconds: 2
+          timeoutSeconds: 1
+          periodSeconds: 1
           successThreshold: 1
-          failureThreshold: 5
+          failureThreshold: 1
```

Comment:
I suspect that this is too aggressive. Health checks will occasionally hiccup. Maybe this was just a convenience for testing?

We'll need to tune this based on real-world experience / how the system actually behaves under load, but how about starting with failureThreshold = 3?

Reply:
Indeed this setup is very aggressive, but I wasn't able to figure out a way to avoid database errors at the preferences server, since the preferences server is not able to recover from a request error from the database. I guess that this could be a topic in an arch meeting to know the impact of "read failures" of the database.

It seems that the CouchDB cluster is working right when a node vanishes, but the load balancer of the kubernetes takes some time to realize of, thus the preferences server gets read errors until the load balancer only routes requests to the CouchDB nodes that are alive.

Reply:
Hi @amatas - could you elaborate what you mean by "the preferences server is not able to recover from a request error from the database"? Do you mean that read errors in the preferences server persist after its backing store has been restored? Or do you mean that the read errors occur as long as the backing store is invalid - in the latter case, what behaviour would you consider appropriate?

Reply:
Hi @amb26, I've just sent an [email to the arch list](https://lists.gpii.net/pipermail/architecture/2017-October/004690.html) describing the setup and the result of the tests.
The read errors only occur when one instance of the cluster is not ready, the cluster is restarting or some of the instances have died. The preferences server seems to work properly once the database cluster is online and all the instances are working right.
I was only saying that any error generated at the database will be reflected on the replies of the preferences server. Perhaps this is the expected behavior, and the GPII clients will raise a popup with the error situation, I don't know. I think I'm not sure that I'm a position to suggest an appropriate behavior because I'm not aware of all the architecture decisions and the requirements of the UX team.
Perhaps we can raise this topic a the arch meeting or continue the discussion at the arch mailing list.